test/e2e/nodepool_kv_advanced_multinet_test.go

```diff
@@ -0,0 +1,328 @@
+		},
+	}
+	defaultNodepool.Spec.DeepCopyInto(&nodePool.Spec)
+
+	if nodePool.Spec.Platform.Kubevirt != nil {
+		nodePool.Spec.Platform.Kubevirt.AdditionalNetworks = []hyperv1.KubevirtNetwork{{
+			Name: "default/" + k.infra.NADName(),
+		}}
+		nodePool.Spec.Platform.Kubevirt.AttachDefaultNetwork = pointer.Bool(false)
+	}
+	nodePool.Spec.Replicas = ptr.To(int32(1))
+	return nodePool, nil
+}
+
+func (k KubeVirtAdvancedMultinetTest) SetupInfra(t *testing.T) error {
+	if err := k.infra.CreateOVNKLayer2NAD("default"); err != nil {
+		return err
+	}
+
+	infraClient, err := k.infra.DiscoverClient()
+	if err != nil {
+		return err
+	}
+
+	t.Log("Creating a proxy dnsmasq pod")
+	dnsmasqPod := k.composeDNSMasqPod()
+	if err := infraClient.Create(k.infra.Ctx(), dnsmasqPod); err != nil {
+		return fmt.Errorf("failed creating dnsmasq pod: %w", err)
+	}
+
+	passthroughService := hcpmanifests.IngressDefaultIngressPassthroughService(k.infra.Namespace())
+	passthroughService.Name = fmt.Sprintf("%s-%s",
+		hcpmanifests.IngressDefaultIngressPassthroughServiceName,
+		k.infra.HostedCluster().Spec.Platform.Kubevirt.GenerateID)
+
+	t.Logf("Waiting for the default ingresss service to appear...")
+	findPassthroughServiceRetryConfig := wait.Backoff{
+		Steps:    15,
+		Duration: 5 * time.Second,
+	}
+	if err := retry.OnError(findPassthroughServiceRetryConfig, apierrors.IsNotFound, func() error {
+		return infraClient.Get(k.infra.Ctx(), client.ObjectKeyFromObject(passthroughService), passthroughService)
+	}); err != nil {
+		return err
+	}
+
+	t.Logf("Waiting for dnsmasq pod to have an address...")
+	err = wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		if err := infraClient.Get(k.infra.Ctx(), client.ObjectKeyFromObject(dnsmasqPod), dnsmasqPod); err != nil {
+			return false, err
+		}
+		return dnsmasqPod.Status.PodIP != "", nil
+	},
+	)
+	if err != nil {
+		return err
+	}
+
+	ports := []discoveryv1.EndpointPort{}
+	for _, port := range passthroughService.Spec.Ports {
+		ports = append(ports, discoveryv1.EndpointPort{
+			Name:     &port.Name,
+			Protocol: &port.Protocol,
+			Port:     pointer.Int32(int32(port.TargetPort.IntValue())),
+		})
+	}
+	eps := discoveryv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: k.infra.Namespace(),
+			Name:      "test-e2e-nodepool-kv-advanced-multinet-default-ingress-passthrough-dnsmasq-ipv4",
+			Labels: map[string]string{
+				discoveryv1.LabelServiceName: passthroughService.Name,
+			},
+		},
+		AddressType: discoveryv1.AddressType(corev1.IPv4Protocol),
+		Ports:       ports,
+		Endpoints: []discoveryv1.Endpoint{{
+			Addresses: []string{dnsmasqPod.Status.PodIP},
+			Conditions: discoveryv1.EndpointConditions{
+				Ready:       pointer.Bool(true),
+				Serving:     pointer.Bool(true),
+				Terminating: pointer.Bool(false),
+			},
+		}},
+	}
+	t.Logf("Creating custom default ingress passthrough endpointslice")
+	if err := infraClient.Create(k.infra.Ctx(), &eps); err != nil {
+		return err
```

Comment:
For anyone that breaks this test and is debugging, having the name & namespace of the objects you're creating explicitly stated in the errors helps a lot

Reply:
Will add the context.